portsyncd/linksync.cpp

```diff
@@ -114,33 +115,39 @@ LinkSync::LinkSync(DBConnector *appl_db, DBConnector *state_db) :
             }
         }
 
-        for (idx_p = if_ni.get();
-                idx_p != NULL && idx_p->if_index != 0 && idx_p->if_name != NULL;
-                idx_p++)
+        /* In DPU SONiC this step is not needed as netdevs in Kernel are created in the early stage of the syncd service start, 
+         * when driver is loading. And exist while the driver is not unloaded.
+        */
+        if (g_switchType != "dpu")
```

Comment:
Can we just add the check and return if `dpu`. If other platform require this in future, lets address it accordingly.